src/atoms/Text/Text.native.js

```diff
@@ -0,0 +1,43 @@
+import PropTypes from 'prop-types';
+import styled from 'styled-components/native';
+import theme from '../../tokens/theme';
+import { Text as NativeText } from 'react-native';
+import { flattenArray } from '../../utils/index';
+
+const fontFamilies = {
+  light: `Lato-Regular`,
+  regular: `Lato-Regular`,
+  bold: `Lato-Bold`,
+};
+
+const Text = styled(NativeText)(
+  (props) => `
+  font-family: ${fontFamilies[props.weight]};
+  font-size: ${props.theme.fonts.size[props.size]};
+  letter-spacing: ${props.theme.fonts.letterSpacing[props.letterSpacing]};
+  color: ${props.color || props.theme.colors.shade[800]};
+  text-decoration-line: ${props.underline ? 'underline' : 'none'};
+  align-self: ${props.align};
+`,
+);
+
+Text.propTypes = {
+  children: PropTypes.node.isRequired,
+  weight: PropTypes.oneOf(Object.keys(theme.fonts.weight)),
+  size: PropTypes.oneOf(Object.keys(theme.fonts.size)),
+  letterSpacing: PropTypes.oneOf(Object.keys(theme.fonts.letterSpacing)),
+  color: PropTypes.oneOf(flattenArray(Object.values(theme.colors).map(Object.values))),
```

Comment:
@kamleshchandnani @shridhar52war @shubhnik  I need some feedback/suggestions on this

PropTypes need to understand what are the acceptable values. Color must be a string value that is defined in our design system colors.

https://jsfiddle.net/nz05xutp/8/

Check the fiddle above to understand how it works and what the output is(check the console)

LMK if this is okay or we can do something better

Reply:
LGTM

Reply:
We'll discuss this in person